src/main/java/com/arpnetworking/clusteraggregator/client/HttpSourceActor.java

```diff
@@ -201,30 +210,38 @@ public Receive createReceive() {
                 .build();
     }
 
-    private static ImmutableMultimap<String, String> createHeaderMultimap(final Iterable<HttpHeader> headers) {
+    private static com.arpnetworking.clusteraggregator.models.HttpRequest mapModel(
```

Comment:
This is starting to make me think that we shouldn't use an internal model for processing.  We used an internal model because it was simpler, but now we're starting to look at some more fields of the Request and I think it's time to rethink.  We'll still want to pull out the ByteString for the body, but perhaps we should just pass along the original HttpRequest instead of creating our own.

Reply:
It is passed on to a point. If you mean pulling it into our model as a reference, I don't think that's helpful either. One option is to integrate more thoroughly with the Akka model. That's an option worth discussing, but only once we ditch TCP; while we don't do it well today, the idea of the internal model is to provide an abstraction over protocol and framework. I'm all for moving to HTTP only, but it's not happening until after MAD 2.0 is out imo and will require a similar major version bump of CAGG. If you had something else in mind, let me know.